### PR TITLE
Diet Recommendation for Pregnancy

### LINKED ISSUE #220: Diet Recommendation for Pregnant Women based on several factors
## Project Request
Develop a pregnancy diet recommendation system that suggests suitable food items for pregnant women based on their personal characteristics, dietary restrictions, nutritional needs, and pregnancy stage. The system should handle multi-label classification, as pregnant women may have multiple dietary preferences and nutritional requirements during different stages of pregnancy.

| Field  | Description                       |
| ------ | --------------------------------- |
| Github | Nabanita29              |
| Email  |  nabanita21csu057@ncuindia.edu   |
| Label  | Project Request                   |

*Define You**

- [ ] GSSOC Participant
- [ ] Contributor

# Project Name

Diet recommendation system for pregnancy

## Description
The solution involves collecting a comprehensive dataset with pregnant women's characteristics, pregnancy stage, dietary restrictions, nutritional needs, and food items with nutritional information. The data will be preprocessed to handle missing values, encode categorical variables, and scale numerical features. Multiple multi-label classification models, including Binary Relevance, Classifier Chains, Label Powerset, and Random Forest, will be evaluated and trained on the dataset. Hyperparameter tuning will be performed to enhance model performance.

## Video Links or Support Links

image

### PR DESCRIPTION
**Pull request by Nabanita29**
##  Info about the related issue
- [x] GSSOC Participant
- [x] Contributor

Closes: #220 

My solution involves collecting a comprehensive dataset with pregnant women's characteristics, pregnancy stage, dietary restrictions, nutritional needs, and food items with nutritional information. The data has been preprocessed to handle missing values, encode categorical variables, and scale numerical features. Multiple multi-label classification models, including Random Forest and neural network model, has been evaluated and trained on the dataset. Hyperparameter tuning has also been performed to enhance model performance.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots
image
image
image

### CODE REVIEW

**Comment by Nabanita29** [comment]
Sir I will update the readme and dataset if you can review and let me know if this project will be accepted !
